src/common/normalizers.ts

```diff
@@ -0,0 +1,38 @@
+import { configuration } from '../singletons/configuration.js';
+
+/**
+ * Normalizes the domain.
+ * In strict mode, normalizes the domain to lower case.
+ *
+ * @param domain the domain to normalize.
+ * @returns the normalized domain.
+ */
+function normalizeDomain<T>(domain: T): T {
```

Comment:
Should we allow generic domains? afaik only strings are used, and I think that we should enforce it.